File: ClipboardZanager/Sources/ClipboardZanager.Shared/CloudStorage/CloudFile.cs
using System;

namespace ClipboardZanager.Shared.CloudStorage
{
    /// <summary>
    /// Represents a file in the cloud provided by a <see cref="ICloudStorageProvider"/>.
    /// </summary>
    public class CloudFile
    {
        /// <summary>
        /// Gets or sets the name of the file.
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// Gets or sets the full path to the file.
        /// </summary>
        public string FullPath { get; set; }

        /// <summary>
        /// Gets or sets the date and time of last motification in Utc format on the server.
        /// </summary>
        public DateTime? LastModificationUtcDate { get; set; }
    }
}

File: Android/app/src/main/java/com/etiennebaudoux/clipboardzanager/componentmodel/cloudstorage/CloudFile.java
package com.etiennebaudoux.clipboardzanager.componentmodel.cloudstorage;

import java.util.Date;

/**
 * Represents a file in the cloud provided by a CloudStorageProvider.
 */
public class CloudFile {
    //region Properties

    //region Name

    private String _name;

    /**
     * Gets the name of the file.
     *
     * @return The name of the file.
     */
    public String getName() {
        return _name;
    }

    /**
     * Sets the name of the file.
     *
     * @param name The name of the file.
     */
    public void setName(String name) {
        _name = name;
    }

    //endregion

    //region FullPath

    private String _fullPath;

    /**
     * Gets the full path to the file.
     *
     * @return The full path to the file.
     */
    public String getFullPath() {
        return _fullPath;
    }

    /**
     * Sets the full path to the file.
     *
     * @param fullPath the full path to the file.
     */
    public void setFullPath(String fullPath) {
        _fullPath = fullPath;
    }

    //endregion

    //region LastModificationUtcDate

    private Date _lastModificationUtcDate;

    /**
     * Gets the date and time of last modification in Utc format on the server.
     *
     * @return The date and time of last modification in Utc format on the server.
     */
    public Date getLastModificationUtcDate() {
        return _lastModificationUtcDate;
    }

    /**
     * Sets the date and time of last modification in Utc format on the server.
     *
     * @param lastModificationUtcDate The date and time of last modification in Utc format on the server.
     */
    public void setLastModificationUtcDate(Date lastModificationUtcDate) {
        _lastModificationUtcDate = lastModificationUtcDate;
    }

    //endregion

    //endregion
}
